File: C++ Implementation/simpleQueue.cpp
#include<iostream>
using namespace std;
class squeue{
    int front;
    int rear;
    int size;
    int *arr;
    public:
        squeue(int size){
            this->size = size;
            front=rear=-1;
            arr = new int[size];
        }

        bool isFull(){
            return rear==size-1;
        }

        bool isEmpty(){
            return front==-1;
        }

        void enqueue(int val){
            if(!isFull()){
                rear++;
                arr[rear] = val;
                if(front==-1){
                    front=0;
                }
            } 
            else{
                cout << "Queue is Full" << endl;
            }
        }

        void dequeue(){
            if(!isEmpty()){
                int val = arr[front];
                front++;
            } else{
                cout << "Queue is empty" << endl;
            }
        }

        void display(){
            for(int i=front;i<=rear;i++){
                cout << arr[i] << " ";
            }
            cout << endl;
        }
};
int main()
{
    squeue sq(5);
    sq.enqueue(10);
    sq.enqueue(20);
    sq.enqueue(30);
    sq.enqueue(40);
    sq.enqueue(50);
    sq.display();

    sq.dequeue();
    sq.dequeue();
    sq.display();

    return 0;
}

File: src/com/pratham/StackQueue/SimpleQueue.java
package com.pratham.StackQueue;

public class SimpleQueue {
    static int front = -1;
    static int rear = -1;
    static int size = 5;
    static int[] q = new int[size];

    public static void main(String[] args) {
        // inserting 5 elements
        enqueue(10);
        enqueue(20);
        enqueue(30);
        enqueue(40);
        enqueue(50);

        // displaying
        System.out.println("Displaying after insertion");
        display();

        // deleting 2 elements
        dequeue();
        dequeue();


        // displaying after delete
        System.out.println("Displaying after delete: ");
        for (int i = front; i <= rear; i++) {
            System.out.println(q[i] + " ");
        }
    }

    static void display() {
        for (int i = 0; i <= rear; i++) {
            System.out.println(q[i] + " ");
        }
    }

    static void enqueue(int num) {
        if (!isFull()) {
            rear++;
            q[rear] = num;
            if(front==-1){
                front++;
            }
        } else{
            System.out.println("Queue is full");
        }
    }

    static void dequeue() {
        if (!isEmpty()) {
            int val = q[front];
            front++;
        } else{
            System.out.println("Queue is Empty");
        }
    }

    static boolean isFull() {
        return rear == size - 1;
    }

    static boolean isEmpty() {
        return front == -1;
    }
}
